main.tf

```diff
@@ -111,6 +112,7 @@ resource "azurerm_application_gateway" "ag" {
       port                  = 80
       protocol              = "Http"
       request_timeout       = 30
+      host_name             = backend_http_settings.value.backend_host_name_override
```

Comment:
Depending if test is fine or not, may need to add additional

[host_name](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/application_gateway#host_name) - (Optional) Host header to be sent to the backend servers. Cannot be set if pick_host_name_from_backend_address is set to true.

[pick_host_name_from_backend_address](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/application_gateway#pick_host_name_from_backend_address) - (Optional) Whether host header should be picked from the host name of the backend server. Defaults to false.

Reply:
Added optional pick_host_name_from_backend_address property.  The host_name will not be set if this is set to true.